### DISCUSSION TITLE
Praetor Funding Proposal

### DISCUSSION THREAD

**Discussion by jigar-arc10** [akash-network/community#92] [Governance Proposals]
## Summary
We are proposing a funding request of 50k USD in $AKT from the community pool for our efforts to enhance user experience and increase Akash network adoption.

## Introduction
Providers currently face several challenges that hinder the user experience. These include the lack of content moderation, lease and financial dashboards, and an ideal upgrade path. We propose several solutions to address these challenges to enhance the Akash Network's functionality. We presented to the steering committee on Thursday, Feb 23, 2023.
Here is the attached presentation. 
[Praetor  Akash - Steering Committee Presentation - 23 Feb 2023.pdf](https://github.com/akash-network/community/files/10819309/Praetor.Akash.-.Steering.Committee.Presentation.-.23.Feb.2023.pdf)

## Proposed Solutions

1. Management API/ Moderation API for Content Moderation
To address the issue of content moderation, we propose implementing a management API and Moderation API in the Akash provider service and Praetor. This will give providers better control over their leases and ensure that only moderated content can be hosted on their provider.  This spec is coming from [wg-content-moderation](https://github.com/akash-network/community/tree/main/wg-content-moderation).

2. Lease Dashboard
Currently, there is no lease dashboard, making it difficult for providers to manage leases effectively. To address this issue, we propose implementing a lease dashboard that will show all the current and past leases, allowing providers to track their status and provide the best possible service to their leaseholders. Providers can also close leases and choose not to host leases from specific wallets.

3. Financial Dashboard
We propose to show a financial dashboard to providers, giving them a comprehensive overview of their finances on the platform. The financial dashboard will also show current and upcoming projections based on leases, historic lease financial data, and more. To extract and process this data, we will use Airflow or integrate with indexers like Cloudmos, ensuring efficient data processing and storage.

4. Upgrade to Helm
To provide providers with access to the latest features and upgrades, we propose upgrading the platform to Helm. This will enable existing non-Praetor providers to have Praetor functionalities such as network upgrades, leases, and financial dashboards. This will also enable praetor providers to get insider and vanguard support from the community.

5. Miscellaneous Features
We propose adding additional features such as banning image-based leases, adding persistent storage for existing providers using the dashboard, and continuing the EDU program, which will educate the community about the Akash platform.

## Funding Request
We request funding of USD 50K in $AKT by next month to implement the proposed solutions. We have a timeline for each solution and will work efficiently to implement them. If the community decides to fund Praetor in 2 tranches to lower the burden on the community pool, We will put in two proposals, the first in March and the second at the end of April. Our target is to finish two roadmap items in the first funding request and will finish the other two on the second funding request. We are open to discussion on how to proceed.

#### Roadmap (Timeline)
1. Content Moderation & Lease Dashboard - March 31, 2023
Implement management API in the Akash provider service for content moderation
Implement management API in Praetor for content moderation
Implement a lease dashboard in Praetor
Help providers implement the moderation API in the Akash provider service.

2. Helm Upgrade - April 30, 2023
Install a new provider with Helm.
Upgrade existing providers to Helm.
Provide Praetor functionalities to existing non-Praetor providers.

3. Financial Dashboard - May 31, 2023
Create a financial dashboard for providers.
Show current month and upcoming projections based on leases
Show historic lease financial data
Process financial data securely using Airflow or integrate with indexers like Cloudmos.

4. Existing Provider Feature - June 30, 2023
Integrate moderation API to allow providers to ban image-based leases
Add persistent storage for existing providers using the dashboard
Support existing EDU program participants and create new curriculums

Conclusion:
Implementing these solutions will significantly enhance the Akash Network's functionality and provide a better user experience for our providers. We look forward to your support and cooperation in achieving our goals.

**Comment by figurestudios** [comment]
I approve this if you can include the following partly or fully:

> Currently, there is no lease dashboard, making it difficult for providers to manage leases effectively. To address this issue, we propose implementing a lease dashboard that will show all the current and past leases, allowing providers to track their status and provide the best possible service to their leaseholders. Providers can also close leases and choose not to host leases from specific wallets.

Please also include whitelists for only allowing certain wallets. This, along with the blocklist, can be done with pricing scripts. Providers could trust organizations to only allow people who have completed KYC or is otherwise allowed to deploy (e.g. in the case of having a provider where only students are allowed to deploy to).

> We propose adding additional features such as banning image-based leases, adding persistent storage for existing providers using the dashboard, and continuing the EDU program, which will educate the community about the Akash platform.

Please also include whitelists for only certain images. Note that this must be done transparently, or people will be frustrated for not getting their deployments to work. This is good as some people understandably do not want to host content that is illegal (e.g. piracy).

It could be implemented by for example: 1) creating example SDLs with attributes saying "whitelisted image," 2) having these providers ONLY respond to the tenants requesting to host whitelisted images, and 3) having auditors add new images to the whitelist and 4) have providers double check that the image is whitelisted before pulling.

It is possible that other solutions will make more sense^^ a blacklist is easy to bypass if someone wants to, as they can simply just re-upload the image and deploy again. Whitelists are also not perfect as the images can be changed after being whitelisted if the image is not yet cached on the provider. This issue exists for Flux as we speak, but the solution is easy -- be in control over either the upload, or the content. If it is possible to create a static link (e.g. by using IPFS?) that would work just fine.

**Reply by jigar-arc10** [reply]
Thank you for your valuable input! 

We will implement the whitelist feature as part of our lease dashboard. This will provide an added layer of security and control for providers in managing their leases, allowing them to restrict access to their services to only specific wallets. This feature will significantly enhance the functionality of our lease dashboard and provide our providers with even more flexibility and control.

We will consider this and explore the possibility of implementing whitelists for specific images to ensure compliance and prevent illegal content from being hosted on the Akash platform.

Thank you for helping us improve our platform!

**Comment by zJuuu** [comment]
I support this proposal but assumed that there will be two proposals to lower the burden on the community pool?

**Reply by jigar-arc10** [reply]
Hi, thanks for the supporting our efforts. We have put this discussion around 4 weeks ago and have not received request from community to convert into two tranches, so we decided to move ahead with single payout.

**Comment by SophoclesSummer** [comment]
This proposal was approved on 3/29/2023 by a 93% majority. 
https://www.mintscan.io/akash/proposals/182
